docs/05-plugin-development.md

```diff
@@ -93,18 +93,20 @@ Next Hook: [`resolveId`](guide/en/#resolveid) to resolve each entry point in par
 
-If `false` is returned for `moduleSideEffects` and no other module imports anything from this module, then this module will not be included in the bundle without checking for actual side-effects inside the module. If `true` is returned, Rollup will use its default algorithm to include all statements in the module that have side-effects (such as modifying a global or exported variable). If `"no-treeshake"` is returned, treeshaking will be turned off for this module and it will also be included in one of the generated chunks even if it is empty. If `null` is returned or the flag is omitted, then `moduleSideEffects` will be determined by the first `resolveId` hook that resolved this module, the `treeshake.moduleSideEffects` option, or eventually default to `true`. The `transform` hook can override this.
+If `false` is returned for `moduleSideEffects` and no other module imports anything from this module, then this module will not be included in the bundle even if the module would have side-effects. If `true` is returned, Rollup will use its default algorithm to include all statements in the module that have side-effects (such as modifying a global or exported variable). If `"no-treeshake"` is returned, treeshaking will be turned off for this module and it will also be included in one of the generated chunks even if it is empty. If `null` is returned or the flag is omitted, then `moduleSideEffects` will be determined by the first `resolveId` hook that resolved this module, the `treeshake.moduleSideEffects` option, or eventually default to `true`. The `transform` hook can override this.
 
 See [synthetic named exports](guide/en/#synthetic-named-exports) for the effect of the `syntheticNamedExports` option. If `null` is returned or the flag is omitted, then `syntheticNamedExports` will be determined by the first `resolveId` hook that resolved this module or eventually default to `false`. The `transform` hook can override this.
 
-You can use [`this.getModuleInfo`](guide/en/#thisgetmoduleinfomoduleid-string--moduleinfo) to find out the previous values of `moduleSideEffects` and `syntheticNamedExports` inside this hook.
+See [custom module meta-data](guide/en/#custom-module-meta-data) for how to use the `meta` option. If `null` is returned or the option is omitted, then `meta` will be determined by the first `resolveId` hook that resolved this module or eventually default to an empty object. The `transform` hook can add or replace properties of this object.
```

Comment:
```suggestion
See [custom module meta-data](guide/en/#custom-module-meta-data) for how to use the `meta` option. If a `meta` object is returned by this hook, it will overwrite any `meta` object returned by the `resolveId` hook. If no hook returns a `meta` object it will default to an empty object. The `transform` hook can add or replace properties of this object.
If `null` is returned or the option is omitted, then `meta` will be determined by the first `resolveId` hook that resolved this module or eventually default to an empty object. The `transform` hook can add or replace properties of this object.
```

Reply:
It took me a few reads to understand the meaning. I think it's clearer to say that load overwrites whatever resolve returns.

The overwriting is a bit unfortunate however, would it not be possible to merge the objects?

Reply:
It is not overwritten, the `meta` object is indeed merged shallowly so that different plugins do not interact. Example:

```
// meta after resolveId
{
 'node-resolve': {
    someData: 'foo'
  },
  commonjs: {
    someData: 'bar'
  }
}

// returned meta in load
{
  commonjs: {
    someOtherData: 'baz'
  }
}

// meta after load
{
 'node-resolve': {
    someData: 'foo'
  },
  commonjs: {
    someOtherData: 'baz'
  }
}
```

That is why I wrote
> The `transform` hook can add or replace properties of this object.

So maybe we need to write this in yet another way?

Reply:
How about

> If a `meta` object is returned by this hook, it will be merged shallowly with any `meta` object returned by the `resolveId` hook. If no hook returns a `meta` object it will default to an empty object. The `transform` hook can add or replace properties of this object.

Reply:
I added my change to the docs.